### PR TITLE
MGMT-17065 Add Enrollment request details page

### PR DESCRIPTION
**Pull request by rawagner**
er_details

### CODE REVIEW

File: src/app/components/DetailsPage/ConditionsTable.tsx [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 type ConditionsTableProps = {
-  device: Device;
+  conditions: Array<Condition> | undefined;
 };
 
-const ConditionsTable: React.FC<ConditionsTableProps> = ({ device }) => {
-  return device.status?.conditions?.length ? (
+const ConditionsTable: React.FC<ConditionsTableProps> = ({ conditions }) => {
+  return conditions?.length ? (
     <Table aria-label="Device conditions table">
```
Comment (celdrake) [inline src/app/components/DetailsPage/ConditionsTable.tsx:12]: We could probably add a "type" prop and use it for the `aria-label`, and also for the empty state text.

```suggestion
    <Table aria-label={`${type} conditions table`}>
```

**Review by celdrake — CHANGES_REQUESTED** [review]

File: src/app/components/DetailsPage/IntegrityTable.tsx [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
 
 type IntegrityTableProps = {
-  device: Device;
+  measurements: Record<string, string> | undefined;
 };
 
-const IntegrityTable: React.FC<IntegrityTableProps> = ({ device }) => {
-  return device.status?.systemInfo?.measurements ? (
+const IntegrityTable: React.FC<IntegrityTableProps> = ({ measurements }) => {
+  return measurements ? (
```
[the remark this answers is not in the document]
Comment (rawagner) [reply src/app/components/DetailsPage/IntegrityTable.tsx]: fixed

File: src/app/components/DetailsPage/ConditionsTable.tsx [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
 import * as React from 'react';
 
 type ConditionsTableProps = {
-  device: Device;
+  conditions: Array<Condition> | undefined;
 };
 
-const ConditionsTable: React.FC<ConditionsTableProps> = ({ device }) => {
-  return device.status?.conditions?.length ? (
+const ConditionsTable: React.FC<ConditionsTableProps> = ({ conditions }) => {
```
[the remark this answers is not in the document]
Comment (rawagner) [reply src/app/components/DetailsPage/ConditionsTable.tsx]: I've added `message` column to the table

File: src/app/utils/status/enrollmentRequest.ts [hunk 4]
```diff
@@ -1,8 +1,8 @@
 import { ConditionType, EnrollmentRequest } from '@types';
 import { EnrollmentRequestApprovalStatus } from '@app/types/extraTypes';
 
-export const getApprovalStatus = (enrollmentRequest: EnrollmentRequest): EnrollmentRequestApprovalStatus => {
-  const approvedCondition = enrollmentRequest.status?.conditions?.find(
+export const getApprovalStatus = (enrollmentRequest?: EnrollmentRequest): EnrollmentRequestApprovalStatus => {
```
[the remark this answers is not in the document]
Comment (rawagner) [reply src/app/utils/status/enrollmentRequest.ts]: Sounds good, new code is 
```
const approvalStatus = er ? getApprovalStatus(er) : '-'; 
```

File: src/app/components/DetailsPage/ConditionsTable.tsx (hunk 1 above)
Comment (rawagner) [reply src/app/components/DetailsPage/ConditionsTable.tsx]: fixed

File: src/app/components/EnrollmentRequest/EnrollmentRequestDetails/EnrollmentRequestDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+                      value={er.spec.csr}
+                      readOnlyVariant="plain"
+                      autoResize
+                      className="fctl-enrollment-details__text-area"
+                    />
+                  ) : (
+                    <Bullseye>Not available</Bullseye>
+                  )}
+                </DetailsPageCardBody>
+                <CardFooter>A PEM-encoded PKCS#10 certificate signing request.</CardFooter>
```
[the remark this answers is not in the document]
Comment (rawagner) [reply src/app/components/EnrollmentRequest/EnrollmentRequestDetails/EnrollmentRequestDetails.tsx]: I've moved the description to a popover content, next to title. Let me know what you think.

We will always have CSR,
Certificate will be there once the device is approved. I woudn't want to hide the card completely as it changes the page layout quite a bit. Maybe we could merge these two cards into one ? Or just wait for proper UX design :)

[the remark this answers is not in the document]
Comment (celdrake) [reply src/app/components/EnrollmentRequest/EnrollmentRequestDetails/EnrollmentRequestDetails.tsx]: I like it better now. Of course it's temporary, but I think it looks nicer, thanks!

**Review by celdrake — APPROVED** [review]

**Pull request merged**